### PR TITLE
Documentation for R4 Immunization Read and Search

### PR DESCRIPTION
**Pull request by Likitha-Seeram**
Added documentation for R4 Immunization read and search features

### CODE REVIEW

File: content/millennium/r4/medications/immunization.md [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+
+## Retrieve by id
+
+List an individual Immunization by its id:
+
+    GET /Immunization/:id
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, patient: true, system: true) %>
```
Comment (ebaranowski) [inline content/millennium/r4/medications/immunization.md]: patient is false

File: content/millennium/r4/medications/immunization.md [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+## Search
+
+Search for Immunizations that meet supplied query parameters:
+
+    GET /Immunization?:parameters
+
+### Authorization Types
+
+<%= authorization_types(practitioner: true, patient: true, system: true) %>
```
Comment (ebaranowski) [inline content/millennium/r4/medications/immunization.md]: patient is false

File: lib/resources/example_json/r4_examples_immunization.rb [hunk 3]
```diff
@@ -0,0 +1,189 @@
+module Cerner
+  module Resources
+
+    R4_IMMUNIZATION_BUNDLE ||=
+      {
```
Comment (ebaranowski) [inline lib/resources/example_json/r4_examples_immunization.rb]: Couple things.

1. The format of the JSON doesnt look right. It should look like Allergy's
2. You could get rid of about 80 lines if you follow Allergy's pattern with the entry
3. I would recommend renaming the 1 as entry
4. Pretty sure this isnt in Sandbox yet. So you will need to update this request accordingly once it is in Sandbox

File: lib/resources/r4/immunization.yaml [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+- name: reportOrigin
+  required: 'No'
+  cardinality: 0..1
+  type: CodeableConcept
+  description: The source of an immunization record when not reported by the person who administered the vaccine.
+  binding:
+    description: The source of an immunization record when not reported by the person who administered the vaccine.
+    terminology:
+    - display: Immunization Report Origin
+      system: http://hl7.org/fhir/R4/codesystem-immunization-origin.html
```
Comment (ebaranowski) [inline lib/resources/r4/immunization.yaml]: Pretty sure it should be this: http://terminology.hl7.org/CodeSystem/immunization-origin

File: content/millennium/r4/medications/immunization.md (hunk 2 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/e838b5a0f5c3f4ddb05c041755cfa578fb24972e#diff-3e96067fd99a2fee85880712d23a6719R53

File: content/millennium/r4/medications/immunization.md (hunk 1 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/e838b5a0f5c3f4ddb05c041755cfa578fb24972e#diff-3e96067fd99a2fee85880712d23a6719R93

File: lib/resources/example_json/r4_examples_immunization.rb (hunk 3 above)
Comment (Likitha-Seeram) [reply lib/resources/example_json/r4_examples_immunization.rb]: Updated Json format correctly. And I changed the file to follow the pattern like that of Allergy Intolerance.
https://github.com/cerner/fhir.cerner.com/pull/255/commits/e838b5a0f5c3f4ddb05c041755cfa578fb24972e

I will make sure to make any required updates after validating in sandbox.

File: content/millennium/r4/medications/immunization.md (hunk 1 above)
Comment (ebaranowski) [reply content/millennium/r4/medications/immunization.md]: +1

File: content/millennium/r4/medications/immunization.md (hunk 2 above)
Comment (ebaranowski) [reply content/millennium/r4/medications/immunization.md]: +1

File: lib/resources/example_json/r4_examples_immunization.rb (hunk 3 above)
Comment (ebaranowski) [reply lib/resources/example_json/r4_examples_immunization.rb]: +1

File: lib/resources/r4/immunization.yaml (hunk 4 above)
Comment (Likitha-Seeram) [reply lib/resources/r4/immunization.yaml]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/3fa313c0f09253444d22d4ea494e61e108042e75#diff-2c50e8f41daaba3a640861e7b6e81488L49

File: content/millennium/r4/proprietary-codes.md [hunk 5]
```diff
@@ -190,6 +203,28 @@ This code set is client definable for different patient types such an Outpatient
       "userSelected": true
     }
 
+#### Code Set 72 Clinical Event Codes
+
+This code set is extremely large and highly customized by clients. It maintains all the various clinical events such as vital signs, laboratory results, documents, and immunizations.
+
+    { 
+      "system": "https://fhir.cerner.com/2c400054-42d8-4e74-87b7-80b5bd5fde9f/codeSet/72",
```
Comment (Likitha-Seeram) [inline content/millennium/r4/proprietary-codes.md]: Changed the tenant key in the URL's everywhere applicable to <EHR Source id>
https://github.com/cerner/fhir.cerner.com/pull/255/commits/3fa313c0f09253444d22d4ea494e61e108042e75#diff-f1a7812a3d6e9f4049fb6005eecfb1f5L211

File: lib/resources/r4/immunization.yaml (hunk 4 above)
Comment (ebaranowski) [reply lib/resources/r4/immunization.yaml]: Is it not returning the terminology link? it should

File: content/millennium/r4/proprietary-codes.md (hunk 5 above)
Comment (ebaranowski) [reply content/millennium/r4/proprietary-codes.md]: +1

File: content/millennium/r4/medications.md [hunk 6]
```diff
@@ -0,0 +1,7 @@
+---
+title: Medications | R4 API
+layout: api
+---
+
+<%= render '/medications.*' %>
```
Comment (MaxPhilips) [inline content/millennium/r4/medications.md]: this is going to be a problem. this pulls in content from layouts/medications.html. all the other layouts are resource- and fhir-version-agnostic, but that one references specific resources and links to specific versions of hl7.org

we'll need to possibly split it into medications_r4.html and update its content appropriately

File: content/millennium/r4/medications/immunization.md [hunk 7]
```diff
@@ -0,0 +1,118 @@
...
+
+The following fields are returned if valued:
+
+* [Id](http://hl7.org/fhir/R4/resource-definitions.html#Resource.id){:target="_blank"}
+* [Status](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.status){:target="_blank"}
+* [Status reason](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.statusReason){:target="_blank"}
+* [Vaccine administered](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.vaccineCode){:target="_blank"}
+* [Patient](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.patient){:target="_blank"}
+* [Patient encounter](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.encounter){:target="_blank"}
+* [Administration date](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.occurrence_x_){:target="_blank"}
```
Comment (MaxPhilips) [inline content/millennium/r4/medications/immunization.md]: Administration date/time

File: content/millennium/r4/medications/immunization.md [hunk 8]
```diff
@@ -0,0 +1,118 @@
...
+* [Indicates if the record is from the source who administered the vaccine](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.primarySource){:target="_blank"}
```
Comment (MaxPhilips) [inline content/millennium/r4/medications/immunization.md]: maybe:

Whether the record is from the source that administered the vaccine

File: content/millennium/r4/medications/immunization.md [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+* [Vaccine manufacturer](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.manufacturer){:target="_blank"}
+* [Vaccine lot number](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.lotNumber){:target="_blank"}
+* [Vaccine expiration date](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.expirationDate){:target="_blank"}
+* [Vaccine administration site](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.site){:target="_blank"}
+* [Vaccine administration route](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.route){:target="_blank"}
+* [Dosage administered](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.doseQuantity){:target="_blank"}
+* [Performer](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.performer){:target="_blank"}
+    * [Type of performer (ordering or administering)](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.performer.function){:target="_blank"}
+    * [Who performed](http://hl7.org/fhir/R4/immunization-definitions.html#Immunization.performer.actor){:target="_blank"}
```
Comment (MaxPhilips) [inline content/millennium/r4/medications/immunization.md]: Individual who performed

File: content/millennium/r4/medications/immunization.md [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+### Parameters
+
+ Name      | Required?          | Type          | Description
+-----------|--------------------|---------------|-----------------------------------------------------------------------------------------------------
+ `_id`     | This, or `patient` | [`token`]     | The logical resource id associated with the resource.
+ `patient` | This, or `_id`     | [`reference`] | The patient for the vaccination record. Example: `12345`
+ `date`    | N                  | [`date`]      | Date range into which the immunization administration date falls. Must be prefixed by 'ge' or 'le'.
+
+Notes:
+ - The `date` parameter value should be prefixed once by 'ge' representing the earliest date, and once by 'le' representing the latest date. Examples: `date=ge2015-01-01&date=le2016-01-01`, `date=ge2015-01-01`, `date=le2016-01-01`
```
Comment (MaxPhilips) [inline content/millennium/r4/medications/immunization.md]: need to remove the leading space before the dash character so this is rendered as a bullet

can you switch the dash to an asterisk?

can you fix this in content/millennium/dstu2/medications/immunization.md too?

Comment (MaxPhilips) [inline content/millennium/r4/medications/immunization.md]: need to explain that date can be provided once for either prefix or twice with both prefixes. check out how Appointment does it in dstu2

File: lib/resources/example_json/r4_examples_immunization.rb [hunk 11]
```diff
@@ -0,0 +1,109 @@
...
+      ],
+      "entry": [
+        {
+          "fullUrl": "https://fhir-open.devcerner.com/r4/2c400054-42d8-4e74-87b7-80b5bd5fde9f/Immunization/M43434511",
+          "resource": R4_IMMUNIZATION_ENTRY
+        }
+      ]
+    }
+  end
+end
```
Comment (MaxPhilips) [inline lib/resources/example_json/r4_examples_immunization.rb]: add a newline to this and lib/resources/r4/immunization.yaml

File: lib/resources/r4/immunization.yaml (hunk 4 above)
Comment (Likitha-Seeram) [reply lib/resources/r4/immunization.yaml]: Yes. Both of the following terminology links are not returning anything.
https://terminology.hl7.org/CodeSystem/immunization-origin
https://terminology.hl7.org/CodeSystem/v3-ActReason

File: content/millennium/r4/medications/immunization.md (hunk 8 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: I followed the content from this JIRA: https://jira2.cerner.com/browse/OPENSVC-8398

File: content/millennium/r4/medications.md (hunk 6 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications.md]: Changed it to render new layout https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-5981708289046f1d63210fd0fbff1a7eR6

For the new layout, I used the same content but changed the link to point to r4. https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-d38f08e30b722da7ec389eb023068447R1

File: content/millennium/r4/medications/immunization.md (hunk 7 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-d38f08e30b722da7ec389eb023068447R1

File: content/millennium/r4/medications/immunization.md (hunk 9 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-3e96067fd99a2fee85880712d23a6719R38

File: content/millennium/r4/medications/immunization.md (hunk 10 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Fixed in both the places.

R4: https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-3e96067fd99a2fee85880712d23a6719R65

DSTU2: https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-7154922f6909b151245502b6c0e575f5R59

Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-3e96067fd99a2fee85880712d23a6719R65

File: lib/resources/example_json/r4_examples_immunization.rb (hunk 11 above)
Comment (Likitha-Seeram) [reply lib/resources/example_json/r4_examples_immunization.rb]: Added. https://github.com/cerner/fhir.cerner.com/pull/255/commits/7f7bd5fd0fd7d7d18677ca1d5618107ccb707330#diff-b4a3f148f7c74d28ca272418093bb464R109

File: content/millennium/r4/medications/immunization.md (hunk 8 above)
Comment (MaxPhilips) [reply content/millennium/r4/medications/immunization.md]: that will be fine

File: content/millennium/r4/medications/immunization.md (hunk 9 above)
Comment (MaxPhilips) [reply content/millennium/r4/medications/immunization.md]: needs a lowercase 'w'

File: content/millennium/r4/medications/immunization.md (hunk 10 above)
Comment (MaxPhilips) [reply content/millennium/r4/medications/immunization.md]: can you add the second period to the 'e.g.'s?

File: content/millennium/r4/medications.md (hunk 6 above)
Comment (MaxPhilips) [reply content/millennium/r4/medications.md]: need a trailing newline on that new template file

this url http://hl7.org/fhir/r4/resourceguide.html#3.1.2.3 doesn't lead anywhere (there's no 3.1.2.3 anchor on the page) - the r4 resource guide page is less comprehensive than the dstu2 one. I don't know if/where the r4 version of that content is

we also can't include the medorder vs. medstatement content - we don't expose either of those resources for r4 yet

Comment (Likitha-Seeram) [reply content/millennium/r4/medications.md]: Discussed offline. Updated the new file.
https://github.com/cerner/fhir.cerner.com/pull/255/commits/b519a215146c45c38295cd8fae4b6bbf6b7363af#diff-d38f08e30b722da7ec389eb023068447L3

File: content/millennium/r4/medications/immunization.md (hunk 9 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/b519a215146c45c38295cd8fae4b6bbf6b7363af#diff-3e96067fd99a2fee85880712d23a6719R38

File: content/millennium/r4/medications/immunization.md (hunk 10 above)
Comment (Likitha-Seeram) [reply content/millennium/r4/medications/immunization.md]: Sorry for missing that. Added it.
https://github.com/cerner/fhir.cerner.com/pull/255/commits/b519a215146c45c38295cd8fae4b6bbf6b7363af#diff-3e96067fd99a2fee85880712d23a6719L66

File: lib/resources/r4/immunization.yaml (hunk 4 above)
Comment (MaxPhilips) [reply lib/resources/r4/immunization.yaml]: @Likitha-Seeram change the 'system' to 'http://terminology.hl7.org/CodeSystem/immunization-origin'

Comment (Likitha-Seeram) [reply lib/resources/r4/immunization.yaml]: Updated. https://github.com/cerner/fhir.cerner.com/pull/255/commits/8ef53a1253acad23197ba1804453f7548b71bad5#diff-2c50e8f41daaba3a640861e7b6e81488L49

Comment (ebaranowski) [reply lib/resources/r4/immunization.yaml]: +1

**Pull request merged**